File: CIS/leapyear.c
#include <stdio.h>
int leapyear(int year)
{
    if((year%4==0 && year%100!=0) || year%400==0)
    {
        return 1;
    }
    else
    {
        return 0;
    }
    
}
int main()
{
    
    int days;
    int year=1980;
    scanf("%d",&days);
    while (days>365)
    {
        if(leapyear(year))
        {
        
           days-=366;
           if(days==0)
           {
               year=year;
           }
           else
           {
               year+=1;
           }
           
           
            
        }
        else
        {
            
            days-=365;
            year+=1;
               
        }
        

    }
    printf("the year is %d",year);
    return 0;

}

File: java/leapyear.java
import java.util.Scanner;
public class leapyear {
    public static void main(String[] args) {
        Scanner input=new Scanner (System.in);
        System.out.print("enter a year: ");
        int year =input.nextInt();
        boolean isleapyear=(year%4==0&&year%100!=0)||(year%400==0);
        System.out.println(year+" is a leapyear: "+isleapyear);
    }
}